### PR TITLE
Dev/current

### PR DESCRIPTION
**Pull request by cmauf**

### CODE REVIEW

File: docs/components/course/course-nav/course-nav-chapter.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+CourseNavChapter
+===============
+This component is used to display the follow edit view of a course nav.
+## Structure
+
+### Roots
+- [`CourseNavigationEditor`](...)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-chapter.md]: correct linkage would be (../course-nav)

File: docs/components/course/course-nav/course-nav-chapter.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+### Dependencies
+* `Draggable`from`vuedraggable`
+* `chapterFollowSet`from`@/mixins/general/course-chapters`
+* `deepCopy`from`@/mixins/general/helpers`
+* `v4 as uuidv4`from`uuid`
+
+### Components
+- [`CourseNavItem`](...)
+- [`CourseNavPropertyEdit`](...)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-chapter.md]: links missing

File: docs/components/course/course-nav/course-nav-chapter.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+* `courseNav`
+* `locale`
+
+Component Functionality
+---------
+### Props
+- `chapterNameDuplicate`: type: Boolean, default set to false
+- `chapterName`: type: String, default set to `chapterName`
+- `chapter`: type: Object, required: true
+- `courseEnd`: type: String, required: true
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-chapter.md]: `required` would be enough; props either need to be required or have a default set

File: docs/components/course/course-nav/course-nav-follow-edit.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+CourseNavFollowEdit
+===============
+This component is used to display the follow edit view of a course nav item.
+
+## Structure
+
+### Components
+- [`SuggestingInput`](..`@/components/helpers/suggesting-input.vue`)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-follow-edit.md]: link is not correct

File: docs/components/course/course-nav/course-nav-item.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+CourseNavItem 
+===============
+This component is used to render a single item in the course navigation editor.
+
+## Structure
+
+### Roots
+- [`CourseNavChapter`](...)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-item.md]: link missing

File: docs/components/course/course-nav/course-nav-item.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+### Dependencies
+* `chapterFollowSet`from`@/mixins/general/course-chapters`
+* `deepCopy`from`@/mixins/general/helpers`
+
+### Components
+- [`CourseNavPropertyEdit`](...)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-item.md]: link missing

File: docs/components/course/course-nav/course-nav-property-edit.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
+CourseNavPropertyEdit
+===============
+Editable property for course nav item .
+
+## Structure
+
+### Roots
+- [`CourseNavChapter`](...)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-property-edit.md]: link missing

File: docs/components/course/course-nav/course-nav.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+## Structure
+
+### Dependencies
++ `courseContentIdGet`from`@/mixins/general/course-structure`
+* `deepCopy`, `stripKey`from`@/mixins/general/helpers`
+* `slugify`from`@/mixins/general/slugs`
+* `v4 as uuidv4`from`uuid`
+
+### Components
+- [`CourseNavChapter`](../accessibility-settings)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav.md]: wrong link

File: docs/components/helpers/suggesting-input.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
+SuggestingInput
+===============
+This component is used to implement a suggesting input box.
+
+## Structure
+
+### Roots
+- [`CourseNavFollowEdit`](..`@/components/course/course-nav/course-nav-follow-edit.vue`)
```
Comment (cmauf) [inline docs/components/helpers/suggesting-input.md]: link is incorrect

**Review by cmauf — COMMENTED** [review]
@Laudablerug please fix links

File: docs/mixins/vue/array.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+Array
+===============
+This component is to array manipulation methods.
```
Comment (cmauf) [inline docs/mixins/vue/array.md]: it is a mixin

File: docs/mixins/vue/array.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+
+## Structure
+
+Component Functionality
+---------
+### Methods
+- `Add`: to add new  items to items
```
Comment (cmauf) [inline docs/mixins/vue/array.md]: method name wrong, typo in description

File: docs/mixins/vue/array.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+===============
+This component is to array manipulation methods.
+
+## Structure
+
+Component Functionality
+---------
+### Methods
+- `Add`: to add new  items to items 
+- `remove`: to remove item at position idx
```
Comment (cmauf) [inline docs/mixins/vue/array.md]: method name wrong, typo in description

File: docs/mixins/vue/list-sort.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
+ListSort
+===============
+This component is to define methods for sorting lists
```
Comment (cmauf) [inline docs/mixins/vue/list-sort.md]: it is a mixin

File: docs/mixins/vue/list-sort.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+
+## Structure
+
+### Dependencies
+* `local` from `../../assets/i18n`
```
Comment (cmauf) [inline docs/mixins/vue/list-sort.md]: typo

File: docs/mixins/vue/locale.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
+Locale
+===============
+This component is to expose i18n local to components
```
Comment (cmauf) [inline docs/mixins/vue/locale.md]: typo

File: docs/mixins/vue/locale.md [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+---------
+
+### Computed Properties
+#### VueX Getters
+- `profileLanguage`
+- `courseLanguage`
+
+#### Local
+- `courseLangIsAr`: checks if Arabic script language is set, for rtl shenanigans 
+- `i18n`: return local for user language property
```
Comment (cmauf) [inline docs/mixins/vue/locale.md]: typo

File: docs/mixins/vue/password.md [hunk 17]
```diff
@@ -0,0 +1,21 @@
+Password
+===============
+This component is set passwordOk boolean and return password and passwordRepeat
```
Comment (cmauf) [inline docs/mixins/vue/password.md]: it is a mixin

File: docs/mixins/vue/pulgin-edit.md
Comment (cmauf) [inline docs/mixins/vue/pulgin-edit.md:1]: file name has typo

File: docs/mixins/vue/pulgin-edit.md [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+
+### Dependencies
+* `v4 as uuidv4` from `uuid`
+
+Component Functionality
+---------
+### Props
+- `edit`
+  - `type`: `Boolean`
+  - `required`: boolean set to true
```
Comment (cmauf) [inline docs/mixins/vue/pulgin-edit.md]: `edit` is a prop of type boolean that is required!

File: docs/mixins/vue/pulgin-edit.md [hunk 19]
```diff
@@ -0,0 +1,18 @@
+PluginEdit
+===============
+This component uses mixin to populate plugin data on create, and inject properties for plugin edit component
```
Comment (cmauf) [inline docs/mixins/vue/pulgin-edit.md]: it is a mixin!

File: docs/mixins/vue/pulgin-view.md
Comment (cmauf) [inline docs/mixins/vue/pulgin-view.md:1]: typo in file name

File: docs/mixins/vue/pulgin-view.md [hunk 20]
```diff
@@ -0,0 +1,19 @@
+PluginView
+===============
+This component uses mixin to populate plugin data on create, and inject properties for plugin edit component
```
Comment (cmauf) [inline docs/mixins/vue/pulgin-view.md]: it is a mixin

File: docs/mixins/vue/pulgin-view.md [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+### Props
+- `onFinish`
+  - `type`: `[Function, Array]`
+  - `default`: return void 
+- `viewData`
+  - `type`: `Object`
+  - `default`: return null 
+- `editPreview`
+  - `type`: `Boolean`
+  - `default`: set to false
```
Comment (cmauf) [inline docs/mixins/vue/pulgin-view.md]: we use `prop`: type `xx` `yy`, defaults to void for non-required props

File: docs/mixins/vue/routes.md [hunk 22]
```diff
@@ -0,0 +1,36 @@
+Routes
+===============
+This component is to inject properties for route dependent props
```
Comment (cmauf) [inline docs/mixins/vue/routes.md]: it is a mixin

File: docs/mixins/vue/routes.md
Comment (cmauf) [inline docs/mixins/vue/routes.md:1]: has changed, please review source

File: docs/mixins/vue/store-handler.md [hunk 23]
```diff
@@ -0,0 +1,14 @@
+StoreHandler                   
+===============
+This component contains methods to use store in components
```
Comment (cmauf) [inline docs/mixins/vue/store-handler.md]: it is a mixin

File: docs/mixins/vue/store-handler.md [hunk 24]
```diff
@@ -0,0 +1,14 @@
...
+
+## Structure
+
+### Methods
+- `courseFetch`: get course from database and put in store, start other functions if successful   
+- `enrollmentFetch`: load enrollment status of user 
+- `enrollmentUpdate`: update enrollment in store  
+- `flagsFetch`: Fetch flags for course from vuex
+- `flagsUpdate`: persist flag state in database 
+- `notificationsFetch`: get notifications newer than newest in store
```
Comment (cmauf) [inline docs/mixins/vue/store-handler.md]: if vuex properties are called, name them (just like the line below)

File: docs/mixins/vue/time.md [hunk 25]
```diff
@@ -0,0 +1,12 @@
+Time                   
+===============
+This component provides localized timestamps
```
Comment (cmauf) [inline docs/mixins/vue/time.md]: it is a mixin

File: docs/mixins/vue/time.md [hunk 26]
```diff
@@ -0,0 +1,12 @@
+Time                   
+===============
+This component provides localized timestamps 
+
+## Structure
+
+### Methods
+- `dateLocalized`: localize given date string    
+- `dataShort`: Shorten date to DD-MM-YY 
+- `timeLoclized`: localize given time string
```
Comment (cmauf) [inline docs/mixins/vue/time.md]: typo

File: docs/mixins/vue/tooltip-icon.md [hunk 27]
```diff
@@ -0,0 +1,20 @@
+TooltipIcon
+===============
+This component is to define variables for component tooltips
```
Comment (cmauf) [inline docs/mixins/vue/tooltip-icon.md]: mixin

File: docs/mixins/vue/watch-content.md [hunk 28]
```diff
@@ -0,0 +1,16 @@
+WatchContent
+===============
+This component is to add watcher for content to every view, thus updating when data changes
```
Comment (cmauf) [inline docs/mixins/vue/watch-content.md]: mixin

File: docs/mixins/vue/routes.md [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+  - `default`: return empty string
+- `chapters`
+  - `type`: `String`
+  - `default`: return empty string
+- `slug`
+  - `type`: `String`
+  - `default`: return empty string
+- `step`
+  - `type`: `String`
+  - `default`: return empty string
```
Comment (cmauf) [inline docs/mixins/vue/routes.md]: please review as explained before

File: docs/mixins/vue/routes.md [hunk 30]
```diff
@@ -0,0 +1,36 @@
...
+
+### Computed Properties
+#### VueX Getters
+- `courseContent`
+- `courseContentPathId`
+
+#### Local
+- `contentToDIsplay`: return current content block
```
Comment (cmauf) [inline docs/mixins/vue/routes.md]: typo

**Review by cmauf — COMMENTED** [review]
please review

**Review by cmauf — COMMENTED** [review]
please check for typos and inconsistencies

File: docs/components/course/course-nav/course-nav-chapter.md (hunk 3 above)
Comment (Laudablerug) [reply docs/components/course/course-nav/course-nav-chapter.md]: that means only required without : true?

Comment (cmauf) [reply docs/components/course/course-nav/course-nav-chapter.md]: yes, `- `chapter`: type: Object, required` would suffice

File: docs/components/course/course-nav/course-nav-item.md [hunk 31]
```diff
@@ -5,14 +5,14 @@ This component is used to render a single item in the course navigation editor.
 ## Structure
 
 ### Roots
-- [`CourseNavChapter`](...)
+- [`CourseNavChapter`](../course-nav)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-item.md]: wrong path

File: docs/components/course/course-nav/course-nav-item.md [hunk 32]
```diff
@@ -5,14 +5,14 @@ This component is used to render a single item in the course navigation editor.
...
 
 ### Dependencies
 * `chapterFollowSet`from`@/mixins/general/course-chapters`
 * `deepCopy`from`@/mixins/general/helpers`
 
 ### Components
-- [`CourseNavPropertyEdit`](...)
+- [`CourseNavPropertyEdit`](../course-nav)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-item.md]: wrong path

File: docs/components/course/course-nav/course-nav-property-edit.md [hunk 33]
```diff
@@ -5,7 +5,7 @@ Editable property for course nav item .
 ## Structure
 
 ### Roots
-- [`CourseNavChapter`](...)
+- [`CourseNavChapter`](../course-nav)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav-property-edit.md]: wrong path

File: docs/components/course/course-nav/course-nav.md [hunk 34]
```diff
@@ -10,7 +10,7 @@ This component provides a file wrap course navigation editor.
 * `v4 as uuidv4`from`uuid`
 
 ### Components
-- [`CourseNavChapter`](../accessibility-settings)
+- [`CourseNavChapter`](../course-nav)
```
Comment (cmauf) [inline docs/components/course/course-nav/course-nav.md]: wrong path

File: docs/components/helpers/suggesting-input.md [hunk 35]
```diff
@@ -5,7 +5,7 @@ This component is used to implement a suggesting input box.
 ## Structure
 
 ### Roots
-- [`CourseNavFollowEdit`](..`@/components/course/course-nav/course-nav-follow-edit.vue`)
+- [`CourseNavFollowEdit`](../../course/course-nav)
```
Comment (cmauf) [inline docs/components/helpers/suggesting-input.md]: wrong path

File: docs/mixins/vue/list-sort.md [hunk 36]
```diff
@@ -1,12 +1,11 @@
...
 ===============
-This component is to define methods for sorting lists
+This mixin is to define methods for sorting lists
 
 ## Structure
 
 ### Dependencies
-* `local` from `../../assets/i18n`
-* `vuex`
+* `as local` from `../../assets/i18n`
```
Comment (cmauf) [inline docs/mixins/vue/list-sort.md]: the spelling is still wrong

File: docs/mixins/vue/plugin-edit.md [hunk 37]
```diff
@@ -11,8 +11,8 @@ Component Functionality
 ---------
 ### Props
 - `edit`
-  - `type`: `Boolean`
-  - `required`: boolean set to true
+  - `type`: Object 
+  - `required`:
```
Comment (cmauf) [inline docs/mixins/vue/plugin-edit.md]: document props like in course-nav-chapter

File: docs/mixins/vue/plugin-view.md [hunk 38]
```diff
@@ -9,11 +9,11 @@ Component Functionality
...
-  - `default`: return void 
+  - `default`: void 
 - `viewData`
-  - `type`: `Object`
+  - `type`: Object 
   - `default`: return null 
 - `editPreview`
-  - `type`: `Boolean`
+  - `type`: Boolean
   - `default`: set to false
```
Comment (cmauf) [inline docs/mixins/vue/plugin-view.md]: same, document like in course-nav-chapter

File: docs/mixins/vue/routes.md [hunk 39]
```diff
@@ -11,19 +11,19 @@ Component Functionality
...
+  - `type`: prob
   - `default`: return empty string
 - `slug`
-  - `type`: `String`
+  - `type`: prob
   - `default`: return empty string
 - `step`
-  - `type`: `String`
+  - `type`: prob
   - `default`: return empty string
```
Comment (cmauf) [inline docs/mixins/vue/routes.md]: ?

File: docs/store/modules/applications.md [hunk 40]
```diff
@@ -0,0 +1,118 @@
...
+
+### applicationList: 
+The provided getter, retrieves the applicationList array from the state object.
+- `applicationText`: string
+- `areaOfExpertise`: string
+- `fullName`: string
+- `id`: number
+- `institution`: string
+- `votes?`: number 
+### editorVotes
```
Comment (cmauf) [inline docs/store/modules/applications.md:26]: missing description

File: docs/store/modules/applications.md [hunk 41]
```diff
@@ -0,0 +1,118 @@
...
+- `areaOfExpertise`: string
+- `fullName`: string
+- `id`: number
+- `institution`: string
+- `votes?`: number 
+### editorVotes 
+- `applicationId`: number
+- `editorId`: number
+- `vote`: boolean 
+### editorNumber
```
Comment (cmauf) [inline docs/store/modules/applications.md:31]: missing description

File: docs/store/modules/applications.md [hunk 42]
```diff
@@ -0,0 +1,118 @@
...
+### userApplication
```
Comment (cmauf) [inline docs/store/modules/applications.md:34]: missing description

File: docs/store/modules/applications.md [hunk 43]
```diff
@@ -0,0 +1,118 @@
...
+This function is to set number of editors 
+
+### editorVoteAdd
+This function is to push editorVote into editorVotes array, update application's vote count accordingly 
+
+### editorVoteChange 
+This function is to change value for editor's vote on application, call only if vote for editor already exists
+
+### updatePersistedVote 
+This function is to change state after new vote is saved in db
```
Comment (cmauf) [inline docs/store/modules/applications.md]: some of them take arguments, please document these

**Review by cmauf — COMMENTED** [review]
please update taking comments into account

File: docs/mixins/vue/list-sort.md (hunk 14 above)
Comment (cmauf) [reply docs/mixins/vue/list-sort.md]: it must be `locale`

File: docs/store/modules/authentication.md [hunk 44]
```diff
@@ -0,0 +1,50 @@
...
+===============
+
+This store module is to organize authentications for users, including login, logout, and account verification actions, using the Axios library for HTTP requests. It manages state variables for userId and online status and provides getters to access them.
+
+## Dependencies
+- `http` from `axios`
+
+## State
+- `userId`: initialized to `0`
+- `onLine`: set to false
```
Comment (cmauf) [inline docs/store/modules/authentication.md:11]: typo

**Review by cmauf — COMMENTED** [review]
ok

**Pull request merged**